app/services/foreman/renderer/scope/macros/helpers.rb

```diff
@@ -155,6 +155,32 @@ def truthy?(value = nil)
           def falsy?(value = nil)
             Foreman::Cast.to_bool(value) == false
           end
+
+          apipie :method, 'Generate a web request' do
+            keyword :utility, String, desc: 'The utility to use for the web request ("curl" or "wget").'
+            keyword :url, String, desc: 'The URL for the web request.'
+            keyword :ssl_ca_cert, String, desc: 'The path to the SSL certificate.', default: nil
+            keyword :headers, Array, of: String, desc: 'Array containing the headers for the web request starting with "--header".', default: nil
+            keyword :params, Array, of: String, desc: 'Array containing the POST parameters for the web request.', default: nil
+            keyword :output_file, String, desc: 'The path were the result of the web request will be stored.', default: nil
+            returns String, desc: 'The web request.'
+            example 'generate_web_request(utility: "curl", url: "https://www.example.com/register", ssl_ca_cert: "/etc/ssl/custom_certs/ca_cert.crt", headers: ["--header \'Authorization: Bearer <my_token>\'"], params: ["host[build]=false", "host[organization_id]=1"])'
+            example 'generate_web_request(utility: "curl", url: "https://www.example.com/keys/client.asc", output_file: "/etc/apt/trusted.gpg.d/client1.asc")'
+          end
+          def generate_web_request(utility:, url:, ssl_ca_cert: nil, headers: nil, params: nil, output_file: nil)
+            utility = Foreman.download_utilities.fetch(utility || 'curl')
+            command = ["#{utility[:download_command]} #{url}"]
+            command << "#{utility[:ca_cert]} #{ssl_ca_cert}" if ssl_ca_cert
+            command << utility[:request_type_post] if params && utility[:request_type_post]
+            if output_file
+              command << "#{utility[:output_file]} #{output_file}"
+            elsif utility[:output_pipe]
+              command << utility[:output_pipe]
+            end
+            headers&.each { |header| command << header }
+            utility[:format_params].call(params).each { |param| command << param } if params
```

Comment:
We need to fix this because it generated (in case of curl):
`--data "packages=Paket1\ Paket2" which would mean it tries to install a package with the name "Paket1 Paket2" (which does not work because 2 packages are meant")

Reply:
Thanks Bernhard, I removed the quotes around the post parameters. Tthis made it necessary to escape the ampersands in the `wget` post parameter.
In my tests installing multiple packages during host registration now worked as expected.

Reply:
@ofedoren Any thoughts about the solution with removing the quotes?

Reply:
Maybe I should provide some additional information :sweat_smile: 

Post parameters have to be provided differently for `curl` (one `--data` argument for each post parameter) and `wget` (one `--post-data` argument containing all post parameters). Initially I removed the quotes of the post parameters in the Global Registration template to unify them and when formatting the post parameters for the `curl` or `wget` command respectively I added quotes around each post parameter in the case of `curl` and around the whole `--post-data` argument in the case of `wget`. But by this I added quotes also to those post parameters that where not quoted before (like the `packages` post parameter) which is causing the described problem.
While investigating the problem with installing multiple packages, I was thinking about whether it even makes sense to quote any of the post parameters as the parameters that could cause problems are already escaped. So my new approach is to not quote the parameters and escape them correctly.